src/VisualStudio/CSharp/Impl/Options/Formatting/CodeStylePage.cs

```diff
@@ -11,7 +18,406 @@ internal class CodeStylePage : AbstractOptionPage
+            // csharp_space_between_method_call_parameter_list_parentheses
+            CSharpFormattingOptions_GenerateEditorconfig(optionSet, CSharpFormattingOptions.SpaceWithinMethodCallParentheses, editorconfig);
+            // csharp_space_between_method_declaration_parameter_list_parentheses
+            CSharpFormattingOptions_GenerateEditorconfig(optionSet, CSharpFormattingOptions.SpaceWithinMethodDeclarationParenthesis, editorconfig);
+            // csharp_space_between_parentheses
+            CSharpFormattingOptions_GenerateEditorconfig(optionSet, editorconfig);
+            // csharp_space_before_colon_in_inheritance_clause
+            CSharpFormattingOptions_GenerateEditorconfig(optionSet, CSharpFormattingOptions.SpaceBeforeColonInBaseTypeDeclaration, editorconfig);
+            // csharp_space_after_colon_in_inheritance_clause
+            CSharpFormattingOptions_GenerateEditorconfig(optionSet, CSharpFormattingOptions.SpaceAfterColonInBaseTypeDeclaration, editorconfig);
+            // csharp_space_around_binary_operators
+            CSharpFormattingOptions_GenerateEditorconfig(optionSet, CSharpFormattingOptions.SpacingAroundBinaryOperator, editorconfig);
+            // csharp_space_between_method_declaration_empty_parameter_list_parentheses
+            CSharpFormattingOptions_GenerateEditorconfig(optionSet, CSharpFormattingOptions.SpaceBetweenEmptyMethodDeclarationParentheses, editorconfig);
+            // csharp_space_between_method_call_name_and_opening_parenthesis
+            CSharpFormattingOptions_GenerateEditorconfig(optionSet, CSharpFormattingOptions.SpaceAfterMethodCallName, editorconfig);
+            // csharp_space_between_method_call_empty_parameter_list_parentheses
+            CSharpFormattingOptions_GenerateEditorconfig(optionSet, CSharpFormattingOptions.SpaceBetweenEmptyMethodCallParentheses, editorconfig);
+
+            editorconfig.AppendLine();
+            editorconfig.AppendLine("# Wrapping preferences");
+            // csharp_preserve_single_line_statements
+            CSharpFormattingOptions_GenerateEditorconfig(optionSet, CSharpFormattingOptions.WrappingKeepStatementsOnSingleLine, editorconfig);
+            // csharp_preserve_single_line_blocks
+            CSharpFormattingOptions_GenerateEditorconfig(optionSet, CSharpFormattingOptions.WrappingPreserveSingleLine, editorconfig);
+        }
+
+        private static void CSharpFormattingOptions_GenerateEditorconfig(OptionSet optionSet, Option<bool> option, StringBuilder editorconfig)
+        {
+            var element = (EditorConfigStorageLocation<bool>)option.StorageLocations.OfType<IEditorConfigStorageLocation>().SingleOrDefault();
+            if (element != null)
+            {
+                editorconfig.Append(element.KeyName + " = ");
+
+                editorconfig.AppendLine(optionSet.GetOption(CSharpFormattingOptions.NewLineForElse).ToString().ToLower());
```

Comment:
It's very specific that you have: CSharpFormattingOptions.NewLineForElse here.  Is that intentional?